lambda_backend/predict_microservice/base_inference.py

```diff
@@ -0,0 +1,144 @@
+import cv2
+import numpy as np
+from configparser import RawConfigParser
+
+DEF_SCORE = 0.3
+DEF_NMS = 0.4
+
+class BaseInference:
+    """
+    Base Inference class for predictions
+    
+    ...
+
+    Attributes
+    ----------
+    weight_path : str
+        path to the .weight file
+    config_path : str
+        path to the .cfg file
+    classes : list
+        names of classes detected
+    score_thresh : float
+        threshold to classify object as detected
+    nms_threshold : float
+        threshold for non-max suppression
```

Comment:
Maybe more descriptive names like `nms_confidence_thresh`, `nms_iou_threshold`

Reply:
Doesn't NMS only need IOU threshold? Score threshold is just the minimum for the `Inference` class to classify an object as detected, which means effectively you only need one score threshold, unless I'm mistaken. [They are already filtered within the double for loop of `_get_filtered_boxes`](https://github.com/hughjazzman/RouteMaker/blob/45a77f66f3924739f36729b3a42bc635625aa2f7/lambda_backend/predict_microservice/base_inference.py#L126):

```Python
                if confidence > self.score_thresh:
```

[See also the opencv docs where they are labelled `score_threshold` and `nms_threshold` as well.](https://docs.opencv.org/3.4/d6/d0f/group__dnn.html#ga9d118d70a1659af729d01b10233213ee)